circuit_maintenance_parser/utils.py

```diff
@@ -1,27 +1,116 @@
 """Utility functions for the library."""
+import os
+import logging
+from typing import Tuple
+
+import pandas as pd  # type: ignore
 from geopy.exc import GeocoderUnavailable, GeocoderTimedOut, GeocoderServiceError  # type: ignore
 from geopy.geocoders import Nominatim  # type: ignore
 from tzwhere import tzwhere  # type: ignore
+import backoff  # type: ignore
+
 from .errors import ParserError
 
+logger = logging.getLogger(__name__)
+
+dirname = os.path.dirname(__file__)
+
+
+class Geolocator:
+    """Class to obtain Geo Location coordinates."""
 
-def city_timezone(city: str) -> str:
-    """Get the timezone for a given city.
+    # Keeping caching of local DB and timezone in the class
+    db_location = None
+    timezone = None
 
-    Args:
-        city (str): Geographic location name
-    """
-    try:
+    def __init__(self):
+        """Initialize instance."""
+        self.load_db_location()
+        self.load_timezone()
+
+    @classmethod
+    def load_timezone(cls):
+        """Load the timezone resolver."""
+        if cls.timezone is None:
+            cls.timezone = tzwhere.tzwhere()
+            logger.info("Loaded local timezone resolver.")
+
+    @classmethod
+    def load_db_location(cls):
+        """Load the localtions DB."""
+        if cls.db_location is None:
+            cls.db_location = pd.read_csv(os.path.join(dirname, "data", "worldcities.csv"))
+            logger.info("Loaded local locations DB.")
+
+    def get_location(self, city: str) -> Tuple[float, float]:
+        """Get location."""
+        try:
+            location_coordinates = self.get_location_from_local_file(city)
+        except ValueError:
+            location_coordinates = self.get_location_from_api(city)
+
+        logger.debug(
+            "Resolved city %s to coordinates: lat %s - lon %s", city, location_coordinates[0], location_coordinates[1],
+        )
+        return location_coordinates
+
+    def get_location_from_local_file(self, city: str) -> Tuple[float, float]:
+        """Get location from Local DB."""
+        city_name = city.split(", ")[0]
+        country = city.split(", ")[-1]
+
+        if self.db_location is not None:
```

Comment:
Do we need this check? It doesn't look like `load_db_location()` has any case where it'd fail to populate `self.db_location`.

Reply:
typing check 😓